From 9e63abc9a136ff46991b64ce62aa0e2d70f3c32b Mon Sep 17 00:00:00 2001
From: Manoj Srivastava <srivasta@golden-gryphon.com>
Date: Sat, 27 Apr 2013 17:28:13 -0700
Subject: [topic--destdir]: Add DESTDIR support for all install commands in
 Makefile.am

This is useful by itself, but is pretty ctitical for packaging.

(cherry picked from commit aca847f0ae3c384e21eb7e059224394984decf70)
Signed-off-by: Manoj Srivastava <srivasta@golden-gryphon.com>
---
 Makefile.am       |    2 +-
 icons/Makefile.am |   16 ++++++++--------
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 48c6456..7f83c11 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -14,7 +14,7 @@ SUBDIRS = bin doc icons etc devel rpm
 # is for a distribution package building.
 # Set the default icon set aliases (symlinks).
 install-data-local:
-	$(bindir)/wm-icons-config --user-dir="$(icondir)" --defaults --quiet
+	$(DESTDIR)$(bindir)/wm-icons-config --user-dir="$(DESTDIR)$(icondir)" --defaults --quiet
 
 wm-configs _pack-symlinks _unpack-symlinks:
 	cd devel/bin && $(MAKE) $(AM_MAKEFLAGS) $@
diff --git a/icons/Makefile.am b/icons/Makefile.am
index 9608529..8e07844 100644
--- a/icons/Makefile.am
+++ b/icons/Makefile.am
@@ -16,16 +16,16 @@ EXTRA_DIST = \
 SYMLINKS_LST = _symlinks.lst
 
 install-data-local:
-	$(mkinstalldirs) $(icondir)
+	$(mkinstalldirs) $(DESTDIR)$(icondir)
 	chmod -R o-w,go+rX *
 	cd $(top_srcdir)/icons; \
 	for iconset in $(SELECTED_ICON_SETS); do \
-		rm -rf $(icondir)/$$iconset; \
+		rm -rf $(DESTDIR)$(icondir)/$$iconset; \
 		tar cpf - "$$iconset" \
-			| (cd "$(icondir)" && tar xfBp -); \
-		find "$(icondir)/$$iconset" -name CVS \
+			| (cd "$(DESTDIR)$(icondir)" && tar xfBp -); \
+		find "$(DESTDIR)$(icondir)/$$iconset" -name CVS \
 			-exec rm -r {} \; 2>/dev/null; \
-		(cd "$(icondir)/$$iconset" && [ -f $(SYMLINKS_LST) ] \
+		(cd "$(DESTDIR)$(icondir)/$$iconset" && [ -f $(SYMLINKS_LST) ] \
 			&& $(AWK) '{ system("ln -sf " $$1 " " $$2) }' \
 				< $(SYMLINKS_LST) \
 			&& rm $(SYMLINKS_LST) || true \
@@ -34,9 +34,9 @@ install-data-local:
 
 uninstall-local:
 	for iconset in $(SELECTED_ICON_SETS); do \
-		rm -rf $(icondir)/$$iconset; \
+		rm -rf $(DESTDIR)$(icondir)/$$iconset; \
 	done
-	for i in $(icondir)/*; do \
+	for i in $(DESTDIR)$(icondir)/*; do \
 		if [ -L "$$i" ] && [ ! -e "$$i" ]; then rm -f "$$i"; fi; \
 	done
-	-rmdir $(icondir)
+	-rmdir $(DESTDIR)$(icondir)
